tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,586 @@
+import { BooleanLike, classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { flow } from 'common/fp';
+import { map, filter, sortBy } from 'common/collections';
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Tabs, Stack, Box, Input, NoticeBox, Icon } from '../components';
+import { Window } from '../layouts';
+import { Food } from './PreferencesMenu/data';
+
+const TYPE_ICONS = {
+  'Can Make': 'utensils',
+  [Food.Alcohol]: 'wine-glass',
+  [Food.Breakfast]: 'sun',
+  [Food.Bugs]: 'bug',
+  [Food.Cloth]: 'tshirt',
+  [Food.Dairy]: 'cheese',
+  [Food.Fried]: 'fire',
+  [Food.Fruit]: 'apple-alt',
+  [Food.Gore]: 'skull',
+  [Food.Grain]: 'wheat-awn',
+  [Food.Gross]: 'trash',
+  [Food.Junkfood]: 'pizza-slice',
+  [Food.Meat]: 'bacon',
+  [Food.Nuts]: 'seedling',
+  [Food.Oranges]: 'apple-alt',
+  [Food.Pineapple]: 'apple-alt',
+  [Food.Raw]: 'drumstick-bite',
+  [Food.Seafood]: 'fish',
+  [Food.Sugar]: 'candy-cane',
+  [Food.Toxic]: 'biohazard',
+  [Food.Vegetables]: 'carrot',
+};
```

Comment:
```suggestion
} as const;
```
Since we're making constants